backend/app/testbot/server.py

```
@@ -0,0 +1,45 @@
+                    context_text = "\n".join(context_docs)
+                    prompt = (f"Use the following information to answer the question.\n\n"
+                              f"Context:\n{context_text}\n\nQuestion: {query}\nAnswer:")
+                    response_text = ask_llm(prompt)
+            except json.JSONDecodeError:
+                response_text = "Invalid JSON message received."
+            # Send the answer and an end marker
+            await websocket.send(response_text)
+            await websocket.send("[END]")
+    except websockets.exceptions.ConnectionClosed:
+        print("Client disconnected")
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)